src/mod.ts

```diff
@@ -35,7 +35,9 @@ const startServer = async function (port: number) {
   for await (const conn of server) {
     // In order to not be blocking, we need to handle each connection individually
     // without awaiting the function
-    serveHttp(conn);
+    (async () => {
```

Comment:
I wish I understood why this would do anything different than what it was. Is it worth testing the rest of these changes w/o this change, or did you already try that?

Reply:
I simplified the `async/await` block and updated PR. Latency distribution also looks better now!

```
Before (with anonymous async/await block)
Running 20s test @ http://127.0.0.1:4500/health
  2 threads and 500 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    12.78ms    2.36ms  98.39ms   92.63%
    Req/Sec    19.65k     1.37k   22.10k    74.00%
  Latency Distribution
     50%   12.47ms
     75%   13.41ms
     90%   14.59ms
     99%   17.54ms
  781922 requests in 20.01s, 104.40MB read
  Socket errors: connect 0, read 1352, write 2, timeout 0
Requests/sec:  39072.61
Transfer/sec:      5.22MB
```

```
After (simplifying with just await)
Running 20s test @ http://127.0.0.1:4500/health
  2 threads and 500 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    75.09us  101.01us  24.08ms   97.74%
    Req/Sec    13.03k     2.17k   15.22k    92.50%
  Latency Distribution
     50%   63.00us
     75%   72.00us
     90%  102.00us
     99%  214.00us
  259238 requests in 20.06s, 34.61MB read
  Socket errors: connect 0, read 6185, write 2, timeout 0
Requests/sec:  12921.36
```